client/my-sites/backup/rewind-flow/review-prompt.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+import React, { FunctionComponent, useCallback } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { Button } from '@wordpress/components';
+import { hasReceivedRemotePreferences as getHasReceivedRemotePreferences } from 'calypso/state/preferences/selectors';
+import Gridicon from 'calypso/components/gridicon';
+import QueryPreferences from 'calypso/components/data/query-preferences';
+import RewindFlowNotice, { RewindFlowNoticeLevel } from './rewind-flow-notice';
+import {
+	getIsDismissed,
+	getHasBeenDismissedOnce,
+} from 'calypso/state/jetpack-review-prompt/selectors';
+import {
+	dismissReviewPrompt,
+	dismissReviewPromptPermanently,
+} from 'calypso/state/jetpack-review-prompt/actions';
+
+const JetpackReviewPrompt: FunctionComponent = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const hasReceivedRemotePreferences = useSelector( ( state ) =>
+		getHasReceivedRemotePreferences( state )
+	);
+	const isDismissed = useSelector( ( state ) => getIsDismissed( state ) );
+	const hasBeenDismissedOnce = useSelector( ( state ) => getHasBeenDismissedOnce( state ) );
+
+	const dismiss = useCallback( () => {
+		dispatch( hasBeenDismissedOnce ? dismissReviewPromptPermanently() : dismissReviewPrompt() );
+	}, [ dispatch, hasBeenDismissedOnce ] );
```

Comment:
I'm trying to figure out the logic here..  If the user clicks the "Leave Jetpack a review" a second time, the review prompt is dismissed permanently?  Can you clarify?

Reply:
yeah, I [copied this from some previous work](https://github.com/Automattic/wp-calypso/pull/22735) without enhancing how clear it is:

1. User is shown prompt for the first time
2. User dismisses prompt
3. Preference is set to ms unix time of dismiss
4. If the user goes to restore and 2 weeks have passed review prompt is shown again
5. Dismissing it a second time permanently hides it, never to be shown again ( preference is set to `permanent` )

I am not sure if this is the right approach, but it seems to me we should not constantly badger the people with review prompts.

Reply:
Awesome, thanks for the clarification. 👍  Yes, I agree we should not continually badger the user to submit a review.